api/test/service/store.test.js

```diff
@@ -118,9 +118,161 @@ test('store add returns done if already uploaded', async (t) => {
   // Even if done (CAR already exists in bucket), mapped to user if non existing
   const item = await getItemFromStoreTable(t.context.dynamoClient, alice, link)
   t.truthy(item)
-  t.truthy(item.uploadedAt)
-  t.truthy(item.proof)
-  t.truthy(item.uploaderDID)
+  t.truthy(item?.uploadedAt)
+  t.truthy(item?.proof)
+  t.truthy(item?.uploaderDID)
+})
+
+test('store remove does not fail for non existent link', async (t) => {
+  const server = await createStoreUcantoServer(t.context)
+  const account = alice.did()
+  const data = new Uint8Array([11, 22, 34, 44, 55])
+  const link = await CAR.codec.link(data)
+
+  const request = await CAR.encode([
```

Comment:
What is the reason for using CAR encoder here? You lose the type validation from the capability if you manually create the invocation like this and also have to decode the response. For example, why not do this instead:

```js
import { connect } from '@ucanto/client'
import { CAR, CBOR } from '@ucanto/transport'
import * as StoreCapabilities from '@web3-storage/access/capabilities/store'

const conn = connect({
  id: t.context.serviceDid,
  encoder: CAR,
  decoder: CBOR,
  channel: server
})

const storeRemove = await StoreCapabilities.remove
    .invoke({
      issuer: alice,
      audience: parse(t.context.serviceDid.did()),
      capabilities: [{
        can: 'store/remove',
        with: account,
        nb: { link },
      }],
      proofs: [],
    })
    .execute(conn)
```

Reply:
I know, the blocker for that is how to spin the SST resources in that way of doing tests. I am expecting to use client in integration tests (which performs capability invocation like you are suggesting).

However, for unit tests kind of interaction, we need to interact directly with the server instance and call request after creating it.

Reply:
@vasco-santos i think Alan is saying, you can pass the uncanto server **object** directly to a uncanto client and then use the client to test different scenarios. No http requests occur, not http api gateway needed. The client object uses the server object directly.

Reply:
In that scenario, we get limited by the things we can test on having the client to support them (example: list with pagination, ...)

I think we should use the client, but in integration tests, not unit

Reply:
If you want to test capabilities that are not in the access client _yet_ then you can always create them here in the same way as the access client creates them. You also get the benefit of being able to copy/pasta them to the client after you've defined them here and tested them.

Reply:
Let's sync on this tomorrow and decide

Reply:
i make it so in #23 and i like it